src/Platforms/SQLServerPlatform.php

```diff
@@ -690,6 +697,32 @@ public function getAlterTableSQL(TableDiff $diff)
         return array_merge($sql, $tableSql, $columnSql);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function getRenameTableSQL(string $oldName, string $newName): array
+    {
+        return [
+            sprintf('sp_rename %s, %s', $this->quoteStringLiteral($oldName), $this->quoteStringLiteral($newName)),
+
+            /* Rename table's default constraints names
+             * to match the new table name.
+             * This is necessary to ensure that the default
+             * constraints can be referenced in future table
+             * alterations as the table name is encoded in
+             * default constraints' names. */
+            "DECLARE @sql NVARCHAR(MAX) = N''; "
+                . "SELECT @sql += N'EXEC sp_rename N''' + dc.name + ''', N''' "
+                . "+ REPLACE(dc.name, '" . $this->generateIdentifierName($oldName)
+                . "', "
+                . "'" . $this->generateIdentifierName($newName) . "') + ''', ''OBJECT'';' "
+                . 'FROM sys.default_constraints dc '
+                . 'JOIN sys.tables tbl ON dc.parent_object_id = tbl.object_id '
+                . 'WHERE tbl.name = ' . $this->quoteStringLiteral($newName) . ';'
+                . 'EXEC sp_executesql @sql',
```

Comment:
I think it's fine to have linebreaks inside this string and that this looks a job for nowdoc + sprintf.

```suggestion
            sprintf(
                <<<'SQL'
                DECLARE @sql NVARCHAR(MAX) = N'';
                SELECT @sql += N'EXEC sp_rename N''' + dc.name + ''', N'''
                + REPLACE(dc.name, '%s', '%s') + ''', ''OBJECT'';'
                  FROM sys.default_constraints dc
                  JOIN sys.tables tbl ON dc.parent_object_id = tbl.object_id
                 WHERE tbl.name = %s;
                EXEC sp_executesql @sql'
                SQL,
                $this->generateIdentifierName($oldName),
                $this->generateIdentifierName($newName),
                $this->quoteStringLiteral($newName)
            )
```

Reply:
It was a copy-paste from `getAlterTableSQL()` which actually didn't need to be duplicated. I removed the original code and made the logic of the new method reused.